### PR TITLE
[python][misc] GHA for Python dependency checks

### PR DESCRIPTION
**Pull request by bkmartinjr**
Partial solution for #751

### CODE REVIEW

File: .github/workflows/py-dependency-check.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+# continues to function correctly using the latest upstream dependencies.
+#
+# Method:
+# 1. Check out the repo at the latest release tag
+# 2. Install latest cellxgene-census package from pypi
+# 3. Verify versions match
+# 4. Run tests from the repo
+
+on:
+  schedule:
```
Comment (bkmartinjr) [inline .github/workflows/py-dependency-check.yml:13]: do we want this to run on any other events?

if not, how will we monitor that it is working?  Badge on the REAMDE?

**Review by mlin — APPROVED** [review]
@bkmartinjr Sure, the R version will actually be a little simpler because the release package tarballs include the test code. (a default behavior of the R packager for whatever reason)

**Comment by bkmartinjr** [comment]
@mlin - do you want to add a workflow to this PR, or send me notes on an approach?  I'm fine with either.

File: .github/workflows/py-dependency-check.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+      - name: confirm release tag
+        run: |
+          echo Latest GH release tag ${{ steps.latestrelease.outputs.releasetag }}
+
+      - name: checkout latest release of repo
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ steps.latestrelease.outputs.releasetag }}
+          fetch-depth: 0
```
Comment (atolopko-czi) [inline .github/workflows/py-dependency-check.yml:41]: `0` checks out full history--do we need that? if not, use `1`

**Review by atolopko-czi — CHANGES_REQUESTED** [review]
LGTM, but a few minor points to consider and a suggestion for sending notifications.

File: .github/workflows/py-dependency-check.yml (hunk 1 above)
Comment (bkmartinjr) [reply .github/workflows/py-dependency-check.yml:13]: @pablo-gar - you can make the call. Either in this PR or elsewhere.  This is not the only workflow that needs such a mechanism.

File: .github/workflows/py-dependency-check.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            echo Release tags match
+          fi
+
+      - name: Test with pytest (API, main tests)
+        run: |
+          PYTHONPATH=. pytest -v -rP ./api/python/cellxgene_census/tests/
+
+      - name: Test with pytest (API, experimental)
+        run: |
+          PYTHONPATH=. pytest -v -rP --experimental ./api/python/cellxgene_census/tests/experimental
```
[the remark this answers is not in the document]
Comment (bkmartinjr) [reply .github/workflows/py-dependency-check.yml:76]: I literally implemented "execute same workflow as unit tests".  Do you recall the aforementioned rationale?

File: .github/workflows/py-dependency-check.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies (including experimental)
+        run: |
+          python -m pip install -U pip setuptools wheel
+          pip install -U -r ./api/python/cellxgene_census/scripts/requirements-dev.txt
+          pip install -U cellxgene-census[experimental]
```
[the remark this answers is not in the document]
Comment (bkmartinjr) [reply .github/workflows/py-dependency-check.yml:52]: no.  Not sure why that was done other than it is harmless and extremely cautious

File: .github/workflows/py-dependency-check.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (atolopko-czi) [reply .github/workflows/py-dependency-check.yml]: I wish I remembered! I don't see a need for it now.

**Review by atolopko-czi — APPROVED** [review]
LGTM, and we can add notifications as a follow-up issue, if needed.

File: .github/workflows/py-dependency-check.yml (hunk 1 above)
Comment (pablo-gar) [reply .github/workflows/py-dependency-check.yml:13]: Unless any strong concerns let's go with slack since that's already a convention used by the team

**Comment by pablo-gar** [comment]
@mlin can you complete the R work in a different PR? 

I will add tasks to the original issue #751, one for Python one for R. We can close the ticket once both have been completed.

File: .github/workflows/py-dependency-check.yml (hunk 1 above)
Comment (bkmartinjr) [reply .github/workflows/py-dependency-check.yml:13]: I am unable to figure out how this works with a quick review, so I'm putting this PR on a back-burner for a while.

**Comment by mlin** [comment]
@pablo-gar @bkmartinjr Yes, no problem, I should be able to put in a separate PR for R tomorrow.

**Comment by bkmartinjr** [comment]
After discussion with @pablo-gar, decided to add a status badge and consider slack integration for a later date.

File: .github/workflows/py-dependency-check.yml (hunk 1 above)
Comment (bkmartinjr) [reply .github/workflows/py-dependency-check.yml:13]: added badge to README.md

**Pull request merged**